sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -275,25 +275,33 @@ struct _pi_queue : _pi_object {
   ze_command_queue_handle_t ZeCommandQueue;
 
   // Keeps the PI context to which this queue belongs.
-  pi_context Context;
+  // This field is only set at _pi_queue creation time, and cannot change.
+  // Therefore it can be accessed without holding a lock on this _pi_queue.
```

Comment:
This is not being used, is it?
Can we instead use RW/RO locks?

Reply:
Are you asking whether Context is still used?  I think it s, I just made it const and so it shows up as a deletion and an addition below.  Since it is const it doesn't need to be locked at all, it can never change.  What do you mean by RW/RO lock?

Reply:
The "const" on the members doesn't help from too much locking, does it?
I am looking for something along std::lock_guard vs std::sharded_lock, I guess

Reply:
Yes, const helps a lot.  Any const field cannot be changed after the q object has been created. Since it cannot be changed, you don't need a lock to guard it from changing while it's in use.  So, currently the only thing really locked when locking the _pi_queue are it's non-const members.  ZeCommandQueue could be a const member, except it is reset to nullptr in one place.  And the only other thing that is non-const right now is ZeCommandListFenceMap, but as we add batching support there will be more non-const members.

Reply:
OK, I agree "const" don't need to be locked at all (for some reason I thought we'd lock them with shared_lock)